test/EtherFiOracle.t.sol

```diff
@@ -641,4 +641,33 @@ contract EtherFiOracleTest is TestSetup {
         consensusReached = etherFiOracleInstance.submitReport(report);
         assertTrue(consensusReached); // succeeded
     }
-}
+
+    function test_execute_task_treasury_payout() public {
+        vm.startPrank(superAdmin);
+        roleRegistry.grantRole(etherFiAdminInstance.ETHERFI_ADMIN_ADMIN_ROLE(), chad);
+        vm.startPrank(owner);
+        etherFiOracleInstance.addCommitteeMember(chad);
+        etherFiOracleInstance.setQuorumSize(1);
+        liquidityPoolInstance.setTreasury(address(owner));
+        etherFiAdminInstance.setValidatorTaskBatchSize(10);
+        vm.stopPrank();
+
+        _moveClock(1024 + 2 * slotsPerEpoch);
+
+        IEtherFiOracle.OracleReport memory report = _emptyOracleReport();
+        _initReportBlockStamp(report);
+
+        deal(chad, 1000000 ether);
+        liquidityPoolInstance.deposit{value: 1000000 ether}();
+        report.accruedRewards = 90 ether;
+        report.protocolFees = 10 ether;
+
+        vm.startPrank(chad);
+        etherFiOracleInstance.submitReport(report);
+        skip(1000);
+        etherFiAdminInstance.executeTasks(report);
+        uint256 ownerBal = eETHInstance.balanceOf(address(owner));
+        assertApproxEqAbs(ownerBal, 10 ether, 1);
```

Comment:
wow

Reply:
if would be nicer if we have a unit test to showcase how the off-chain math should work.

```
protocolAccruedRewards = rebase eeth (so includes tnfts eth and bnft eth where bnft owned by liquidity pool)
protocolFees = ethfund eth rewards + treasury rewards + no operator rewards

if 100 eth is earned by etherfi validators
lets z= fraction of bnft held by lp
protocolAccruedRewards = 90 * 29/32 + 3/32*90*(z)
protocolFees = 10 + 3/32*90 * (1-z)

protocolAccruedRewards + protocolFees = 100
```